runtime/server/downloads.go

```diff
@@ -55,12 +57,25 @@ func (s *Server) downloadHandler(w http.ResponseWriter, req *http.Request) {
 	}
 
 	var q runtime.Query
+	var metricsViewName string
+	var filter *runtimev1.MetricsViewFilter
+	timeRange := false
 	switch v := request.Request.(type) {
 	case *runtimev1.ExportRequest_MetricsViewToplistRequest:
 		v.MetricsViewToplistRequest.Limit = int64(request.Limit)
+		metricsViewName = v.MetricsViewToplistRequest.MetricsViewName
+		filter = v.MetricsViewToplistRequest.Filter
+		if v.MetricsViewToplistRequest.TimeStart != nil || v.MetricsViewToplistRequest.TimeEnd != nil {
+			timeRange = true
+		}
 		q, err = createToplistQuery(req.Context(), w, v.MetricsViewToplistRequest, request.Format)
 	case *runtimev1.ExportRequest_MetricsViewRowsRequest:
 		v.MetricsViewRowsRequest.Limit = request.Limit
+		metricsViewName = v.MetricsViewRowsRequest.MetricsViewName
+		filter = v.MetricsViewRowsRequest.Filter
+		if v.MetricsViewRowsRequest.TimeStart != nil || v.MetricsViewRowsRequest.TimeEnd != nil {
+			timeRange = true
+		}
 		q, err = createRowsQuery(req.Context(), w, v.MetricsViewRowsRequest, request.Format)
 	default:
```

Comment:
This will break if we add `case *runtimev1.ExportRequest_TableRows` for example (which we should do at some point, so we can remove the old Parquet export function used by the modeler) – can we move the query-specific checks into the cases and just have `var filename string` or something as output?

Alternatively, we could consider adding an `ExportFilename` function to the `Query` interface?